src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,151 @@
+        });
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User userWithShortPassword = new User();
+        userWithShortPassword.setLogin("validLogin");
+        userWithShortPassword.setPassword("short");
+        userWithShortPassword.setAge(20);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithShortPassword);
+        });
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        User userWithNullPassword = new User();
+        userWithNullPassword.setLogin("validLogin");
+        userWithNullPassword.setPassword(null);
+        userWithNullPassword.setAge(20);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithNullPassword);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User userWithShortLogin = new User();
+        userWithShortLogin.setLogin("log");
+        userWithShortLogin.setPassword("validPassword");
+        userWithShortLogin.setAge(20);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithShortLogin);
+        });
+    }
+
+    @Test
+    void register_minLengthLogin_ok() {
+        User userWithMinLengthLogin = new User();
+
+        String minLengthLogin = "MinLog";
+        userWithMinLengthLogin.setLogin(minLengthLogin);
+        userWithMinLengthLogin.setPassword("validPassword");
+        userWithMinLengthLogin.setAge(20);
+
+        registrationService.register(userWithMinLengthLogin);
+        assertEquals(userWithMinLengthLogin, storageDao.get(minLengthLogin));
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithMinLengthLogin);
+        });
+    }
+
+    @Test
+    void register_emptyPassword_notOk() {
+        User userWithEmptyPassword = new User();
+        userWithEmptyPassword.setLogin("validLogin");
+        userWithEmptyPassword.setPassword("");
+        userWithEmptyPassword.setAge(20);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithEmptyPassword);
+        });
+    }
+
+    @Test
+    void register_ageIsZero_notOk() {
+        User userWithZeroAge = new User();
+        userWithZeroAge.setLogin("validLogin");
+        userWithZeroAge.setPassword("validPassword");
+        userWithZeroAge.setAge(0);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithZeroAge);
+        });
+    }
+
+    @Test
+    void register_ageBelowMinimum_notOk() {
+        User userWithSeventeenYears = new User();
+        userWithSeventeenYears.setLogin("validLogin");
+        userWithSeventeenYears.setPassword("validPassword");
+        userWithSeventeenYears.setAge(17);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithSeventeenYears);
+        });
```

Comment:
The exception type should be specific. Instead of `RuntimeException`, use the specific exception that the `register` method is expected to throw.